api/src/test/java/au/org/aodn/nrmn/restapi/service/SpreadSheetServiceIT.java

```diff
@@ -230,4 +230,18 @@ void duplicateRowsShouldBeRemoved() throws Exception {
         assertEquals(20, validRows.size());
     }
 
+    /** This test is for:
+     * When adding new jobs, only rows (with "total=0" and duplicate more than 3 times) at the end of the sheet should be
+     * removed. Rows (with "total=0" and duplicate more than 3 times) are not at the end of the sheet should be kept.
+     * @throws Exception
+     */
+    @Test
+    void removeBottomDuplicateRowsOnly() throws Exception {
+        FileSystemResource file = new FileSystemResource("src/test/resources/sheets/removeDuplicateBottomRows.xlsx");
+        MockMultipartFile mockFile = new MockMultipartFile("sheets/removeDuplicateBottomRows.xlsx", file.getInputStream());
+        var parsedSheet = sheetService.stageXlsxFile(mockFile, false);
+        var validRows = stagedJobController.getRowsToSave(parsedSheet);
+        assertEquals(487, validRows.size());
+    }
```

Comment:
Please add a few edge case, 

1. The xls file last line is "ID = 486"
2. The xls file last line is "ID = 490"
3. Assert shows that line "ID = 464, 465, 466, 467, 468" exist in the validRows